src/gui/search/searchwidget.cpp

```diff
@@ -132,10 +161,27 @@ namespace
         return tabName;
     }
 
+    nonstd::expected<QStringList, QString> loadHistory(const Path &filePath)
+    {
+        const auto readResult = Utils::IO::readFile(filePath, HISTORY_FILE_MAX_SIZE);
+        if (!readResult)
+        {
+            if (readResult.error().status == Utils::IO::ReadError::NotExist)
+                return {};
+
+            return nonstd::make_unexpected(readResult.error().message);
+        }
+
+        QStringList history;
+        for (const QByteArrayView line : asConst(Utils::ByteArray::splitToViews(readResult.value(), "\n")))
```

Comment:
Can it use `Qt::SkipEmptyParts`?
I intend to make `Qt::SkipEmptyParts` the default mode for `splitToViews()` in a future PR.
The `Qt::KeepEmptyParts` does not adhere to Qt behavior and I would like to deprecate/remove it.

Reply:
> Can it use `Qt::SkipEmptyParts`?
> I intend to make `Qt::SkipEmptyParts` the default mode for `splitToViews()` in a future PR.

So why to change it now?
It is not supposed to contain empty parts (under normal conditions).

Reply:
>It is not supposed to contain empty parts (under normal conditions).

OK. Good to know that it won't be a problem.